packages/react/test/renderers/JsonForms.test.tsx

```diff
@@ -1100,3 +1101,49 @@ test('JsonForms should use react to additionalErrors update', () => {
   expect(wrapper.find('h5').text()).toBe('Foobar');
   wrapper.unmount();
 });
+
+test('JsonForms should used provided middleware if available', () => {
```

Comment:
@sdirix if there are any other scenarios you think would be good to add let me know, this was the most immediate one I could think of

Reply:
Hi! That's already great!

What's tested now is that if the middleware rejects a modification, that then `onChangeHandler` was not called. We should also check that the rendered value did not change.

It would be good to also test the modification case. For example if the middleware modifies an additional value in addition to the regular update and returns that as the new state. Then the `onChange` handler should also be given that modified value and the new value should be rendered.

I think then we have all basic cases covered!

Reply:
I've added a test case for modifying the state value within the middleware, hopefully this covers the modification case. I wasn't fully sure what you meant by `Then the onChange handler should also be given that modified value and the new value should be rendered.`. Wouldn't the onChange handler only be called if we explicitly call it within the middleware?

Reply:
Hi @sdirix 👋, how goes it? Just a reminder that the extra unit test you asked about has been added, and a query as to whether anything else will be needed before this PR can be further progressed?